### PR TITLE
Fixes #2510

### LINKED ISSUE #2510: Pickle of Chem.Mol changes conformations' coordinates
**Description:**

RDKit loses precision in 3D coordinates on pickling. The code below creates a molecule, adds 3D coordinates. It then compares the coordinates of the original molecule vs molecule from multiprocessing. It seems like RDKit saves `float64` into `float` or similar, as pickling/unpickling twice has no effect compared to single pickle/unpickle.

Although the difference is small, it makes calculations irreproducible when one runs processes on a single thread or multiple threads.

It seems like a bug might occur somewhere around this line: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/MolPickler.cpp#L1385

**Example:**
```
from rdkit import Chem
from rdkit.Chem import rdDistGeom
from multiprocessing import Pool
import numpy as np

mol = Chem.MolFromSmiles('CCCOc1cc(Cl)c(Cl)cc1C(O)C1CCCC1')
rdDistGeom.EmbedMolecule(mol)

def g(mol):
    return mol

def compare(mol1, mol2):
    coord1 = mol1.GetConformer(0).GetPositions()
    coord2 = mol2.GetConformer(0).GetPositions()
    return np.abs(coord1 - coord2).max()

# Losing precision during pickling
v1 = g(mol)
v2 = list(Pool(2).map(g, [mol]))[0]
print(compare(v1, v2))

# Repeated pickling does not change coordinates
v3 = list(Pool(2).map(g, [mol]))[0]
print(compare(v2, v3))
```

This code gives:
```
1.7723052092577518e-07
0.0
```

### PR DESCRIPTION
**Pull request by bp-kelley**
Optionally pickle confs as double precision.

### CODE REVIEW

**Comment by bp-kelley** [comment]
@greglandrum I'm unclear if I should rev the pickle version, but I think probably?

File: Code/GraphMol/MolPickler.cpp [hunk 1]
```diff
@@ -929,14 +929,25 @@ void MolPickler::_pickle(const ROMol *mol, std::ostream &ss,
     }
   }
 
-  // pickle the conformations
-  streamWrite(ss, BEGINCONFS);
-  tmpInt = static_cast<int32_t>(mol->getNumConformers());
-  streamWrite(ss, tmpInt);
+  if (propertyFlags & PicklerOps::ConfDouble) {
```
Comment (bp-kelley) [inline Code/GraphMol/MolPickler.cpp]: If you can think of something better than ConfDouble :)

ConfsAsDouble?

Comment (ptosco) [reply Code/GraphMol/MolPickler.cpp]: @bp-kelley `CoordAsDouble`?

**Comment by greglandrum** [comment]
> @greglandrum I'm unclear if I should rev the pickle version, but I think probably?

Yes, I think so.

File: Code/GraphMol/MolPickler.cpp (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/MolPickler.cpp]: How about `CoordsAsDouble`?

File: Code/GraphMol/MolPickler.h [hunk 2]
```diff
@@ -55,7 +55,8 @@ typedef enum {
   PrivateProps = BOOST_BINARY(10000),
   ComputedProps = BOOST_BINARY(100000),
   AllProps =
-      0x7FFFFFFF,  // all data pickled (only 31 bit flags in case enum==int)
+      0x0000FFFF,  // all data pickled (only 31 bit flags in case enum==int)
+  ConfDouble = 0x0001FFFF // save 3D confs as double
```
Comment (greglandrum) [inline Code/GraphMol/MolPickler.h]: It's not just 3D confs.
How about `// save coordinates in conformers as doubles`

File: Code/GraphMol/Wrap/Mol.cpp [hunk 3]
```diff
@@ -248,6 +248,7 @@ struct mol_wrapper {
         .value("PrivateProps", RDKit::PicklerOps::PrivateProps)
         .value("ComputedProps", RDKit::PicklerOps::ComputedProps)
         .value("AllProps", RDKit::PicklerOps::AllProps)
+        .value("ConfDouble", RDKit::PicklerOps::ConfDouble)
```
Comment (greglandrum) [inline Code/GraphMol/Wrap/Mol.cpp]: Now that we know it's possible, I think these should have docstrings (copied from the C++)

File: Code/GraphMol/MolPickler.cpp (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/MolPickler.cpp]: I'd go with `CoordsAsDouble`

File: Code/GraphMol/Wrap/Mol.cpp (hunk 3 above)
Comment (bp-kelley) [reply Code/GraphMol/Wrap/Mol.cpp]: boost::python::enum_ apparently can't handle docstrings

**Comment by greglandrum** [comment]
LGTM, @ptosco ?

**Comment by ptosco** [comment]
@greglandrum Looks good to me too.

**Pull request merged**